include/exec/sequence/transform_each.hpp

```diff
@@ -0,0 +1,204 @@
+        }
+
+        template <same_as<set_value_t> _SetValue, same_as<__t> _Self>
+        friend void tag_invoke(_SetValue, _Self&& __self) noexcept {
+          stdexec::set_value(static_cast<_Receiver&&>(__self.__op_->__receiver_));
+        }
+
+        template <same_as<set_stopped_t> _SetStopped, same_as<__t> _Self>
+          requires __callable<_SetStopped, _Receiver&&>
+        friend void tag_invoke(_SetStopped, _Self&& __self) noexcept {
+          stdexec::set_stopped(static_cast<_Receiver&&>(__self.__op_->__receiver_));
+        }
+
+        template <same_as<set_error_t> _SetError, same_as<__t> _Self, class _Error>
+          requires __callable<_SetError, _Receiver&&, _Error>
+        friend void tag_invoke(_SetError, _Self&& __self, _Error&& __error) noexcept {
+          stdexec::set_error(
+            static_cast<_Receiver&&>(__self.__op_->__receiver_), static_cast<_Error&&>(__error));
+        }
+
+        template <same_as<get_env_t> _GetEnv, __decays_to<__t> _Self>
+        friend env_of_t<_Receiver> tag_invoke(_GetEnv, _Self&& __self) noexcept {
+          return stdexec::get_env(__self.__op_->__receiver_);
+        }
+      };
+    };
+
+    template <class _Sender, class _ReceiverId, class _Adaptor>
+    struct __operation {
+      using _Receiver = stdexec::__t<_ReceiverId>;
+
+      struct __t : __operation_base<_Receiver, _Adaptor> {
+        using __id = __operation;
+        subscribe_result_t<_Sender, stdexec::__t<__receiver<_ReceiverId, _Adaptor>>> __op_;
+
+        __t(_Sender&& __sndr, _Receiver __rcvr, _Adaptor __adaptor)
+          : __operation_base<
+            _Receiver,
+            _Adaptor>{static_cast<_Receiver&&>(__rcvr), static_cast<_Adaptor&&>(__adaptor)}
+          , __op_{exec::subscribe(
+              static_cast<_Sender&&>(__sndr),
+              stdexec::__t<__receiver<_ReceiverId, _Adaptor>>{this})} {
+        }
+
+        friend void tag_invoke(start_t, __t& __self) noexcept {
+          stdexec::start(__self.__op_);
+        }
+      };
+    };
+
+    template <class _Receiver>
+    struct __subscribe_fn {
+      _Receiver& __rcvr_;
+
+      template <class _Adaptor, class _Sequence>
+      auto operator()(__ignore, _Adaptor __adaptor, _Sequence&& __sequence) noexcept(
+        __nothrow_decay_copyable<_Adaptor> && __nothrow_decay_copyable<_Sequence>
+        && __nothrow_decay_copyable<_Receiver>)
+        -> __t< __operation<_Sequence, __id<_Receiver>, _Adaptor>> {
+        return {
+          static_cast<_Sequence&&>(__sequence),
+          static_cast<_Receiver&&>(__rcvr_),
+          static_cast<_Adaptor&&>(__adaptor)};
+      }
+    };
+
+    template <class _Adaptor>
+    struct _NOT_CALLABLE_ADAPTOR_ { };
+
+    template <class _Item>
+    struct _WITH_ITEM_SENDER_ { };
+
+    template <class _Adaptor, class _Item>
+    auto __try_call(_Item*)
+      -> stdexec::__mexception<_NOT_CALLABLE_ADAPTOR_<_Adaptor&>, _WITH_ITEM_SENDER_<_Item>>;
```

Comment:
i think we want to demangle the sender name here.

```suggestion
      -> stdexec::__mexception<_NOT_CALLABLE_ADAPTOR_<_Adaptor&>, _WITH_ITEM_SENDER_<stdexec::__name_of<_Item>>>;
```

Reply:
Good catch